File: pgm/ABC_Java/ABC_296_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        reader.readLine();
        final String S = reader.readLine();
        final String M = "M";
        final String F = "F";
        final String Yes = "Yes";
        final String No = "No";
        System.out.println (
            S.contains (M.concat (M)) || S.contains (F.concat (F)) ? No : Yes);
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_296_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.Scanner
// import java.util.StringTokenizer

// import kotlin.math.*

const val Yes: String = "Yes"
const val No: String = "No"
const val M: String = "M"
const val F: String = "F"
const val MM: String = M + M
const val FF: String = F + F

fun main() {
    readln()
    val S: String = readln()
    if (!S.contains(MM) && !S.contains(FF)) {
        println(Yes)
    } else {
        println(No)
    }
    kotlin.system.exitProcess(0)
}